### PR TITLE
NumPy Advanced Indexing Support

### PR DESCRIPTION
**Pull request by tbennun**
- [x] `ndarray.flat`
- [x] Ellipsis
- [x] `numpy.newaxis`
- [x] int array 1D
- [x] int array ND
- [x] Support literals in list indexing
- [x] bool array indexing for modifications

### CODE REVIEW

File: dace/frontend/python/memlet_parser.py [hunk 1]
```diff
@@ -3,17 +3,27 @@
...
 from dace.memlet import Memlet
-from dace.symbolic import pystr_to_symbolic
+from dace.symbolic import pystr_to_symbolic, SymbolicType
 from dace.frontend.python.common import DaceSyntaxError
 
 MemletType = Union[ast.Call, ast.Attribute, ast.Subscript, ast.Name]
-MemletExpr = namedtuple('MemletExpr', ['name', 'accesses', 'wcr', 'subset'])
+
+
+@dataclass
```
Comment (alexnick83) [inline dace/frontend/python/memlet_parser.py:19]: We should add this to the developer documentation (adds default __init__ and __repr__?)

File: tests/numpy/common.py [hunk 2]
```diff
@@ -138,7 +138,10 @@ def get_rand_arr(ddesc):
                     reference_result = [reference_result]
                     dace_result = [dace_result]
                     for ref, val in zip(reference_result, dace_result):
-                        assert np.allclose(ref, val, equal_nan=True)
+                        if ref.dtype == np.float32:
+                            assert np.allclose(ref, val, equal_nan=True, rtol=1e-3, atol=1e-5)
```
Comment (alexnick83) [inline tests/numpy/common.py:142]: Aren't those error bounds a bit too loose?

**Review by alexnick83 — APPROVED** [review]
Looks fine, but newast has become a monster now.

File: tests/numpy/common.py (hunk 2 above)
Comment (tbennun) [reply tests/numpy/common.py:142]: I took it from another test I had, anyway they were too tight with 1e-8 atol.

File: dace/frontend/python/memlet_parser.py (hunk 1 above)
Comment (tbennun) [reply dace/frontend/python/memlet_parser.py:19]: what do you mean?

Comment (alexnick83) [reply dace/frontend/python/memlet_parser.py:19]: The @dataclass decorator.

Comment (tbennun) [reply dace/frontend/python/memlet_parser.py:19]: Oh, that's a Python built-in now.

**Pull request merged**